src/main/clojure/clara/rules/engine.cljc

```diff
@@ -646,8 +646,8 @@
       (into [type] constraints))))
 
 (defn- join-node-matches
-  [node join-filter-fn token fact env]
-  (let [beta-bindings (try (join-filter-fn token fact {})
+  [node join-filter-fn token fact fact-bindings env]
+  (let [beta-bindings (try (join-filter-fn token fact fact-bindings {})
                            (catch #?(:clj Exception :cljs :default) e
                                (throw-condition-exception {:cause e
```

Comment:
After taking this for a spin (and looking at the unit tests that check for :bindings that broke), I decided to reverse direction and just merge into the existing :bindings field. That preserves the existing behavior in most cases, keeps the structure a bit simpler, and users can easily see where the bindings came from by looking at the rule itself.